File: Java/0094. Binary Tree Inorder Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public void recursiveInorder(TreeNode root, List<Integer> ans) {
        if (root == null) {
            return;
        }
        
        recursiveInorder(root.left, ans);
        ans.add(root.val);
        recursiveInorder(root.right, ans);
    }
    
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> ans = new ArrayList<>();
        recursiveInorder(root, ans);
        return ans;
    }
}

File: CPP/0094. Binary Tree Inorder Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void recursiveInorder(TreeNode* root, vector<int>& ans) {
        if (root == nullptr) {
            return;
        }
        
        recursiveInorder(root->left, ans);
        ans.push_back(root->val);
        recursiveInorder(root->right, ans);
    }
    
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> ans;
        recursiveInorder(root, ans);
        return ans;
    }
};